### ISSUE TITLE
Type declaration issue

### ISSUE THREAD

**Issue by s373r** [mindbrave/uom-ts#1]
Hey!
I'm playing around your superb library but some declaration issues are present:

```ts
import { Unit, div } from "uom-ts";

type Meters = Unit<{m: 1}>;
type Seconds = Unit<{s: 1}>;
type MetersPerSecond = Unit<{m: 1, s: -1}>;

const speed: MetersPerSecond = div(4 as Meters, 2 as Seconds);

console.log(speed);
```

```
sandbox.ts:7:7 - error TS2322: Type 'DivideUnits<AnyUnit, AnyUnit>' is not assignable to type 'Unit<{ m: 1; s: -1; }>'.
  Type 'Number & FilterObjectByValue<_DivideUnits<AnyUnit, AnyUnit>, undefined> & { __exactKeys: string | number; }' is missing the following properties from type '{ m: 1; s: -1; }': m, s

7 const speed: MetersPerSecond = div(4 as Meters, 2 as Seconds);
        ~~~~~
sandbox.ts:7:36 - error TS2345: Argument of type 'Unit<{ m: 1; }>' is not assignable to parameter of type 'AnyUnit'.
  Type 'Unit<{ m: 1; }>' is not assignable to type '{ [key: string]: Exponent; }'.
    Index signature is missing in type 'Number & { m: 1; } & { __exactKeys: "m"; }'.

7 const speed: MetersPerSecond = div(4 as Meters, 2 as Seconds);
                                     ~~~~~~~~~~~
```

Will appreciate any help!

**Comment by mindbrave** [comment]
Hey, it seems there is a problem with newer versions of TS. I'll try to fix that asap so it will work with the latest TS, but for now you can play with TS 3.0. Sorry for inconvenience.

**Comment by s373r** [comment]
> it seems there is a problem with newer versions of TS

I see -- that definitely related 

> but for now you can play with TS 3.0

My project has already migrated to TS 4 beta so I cannot at the moment 

It's okay, I'll continue using the lib just w/ additional boilerplate code as a workaround 

> Sorry for inconvenience

No worries! Please ping me if you will find some solution 

Have a good day, Rafał!

**Comment by mindbrave** [comment]
Hey, I have finally managed to fix that problem, wasn't easy though.
I wanted to keep simplicity of defining your own custom units and that was challenging, fortunately I found a way.
Versions 2.0.0 is out, I changed some things by the way, in example exponents in range <-6, 6> are now supported, I removed curried versions of functions and added overloads for partial application for "ordinary" versions (more of that in readme), also removed need for Scalar unit. TypeScript 4 is now supported.

I hope it will be fine now. Have a good day.
